src/agent/coverage/src/block/linux.rs

```diff
@@ -0,0 +1,507 @@
+    pub loaded: Vec<(u64, ModuleImage)>,
+    pub unloaded: Vec<(u64, ModuleImage)>,
+}
+
+impl LoadEvents {
+    pub fn new(old: &BTreeMap<u64, ModuleImage>, new: &BTreeMap<u64, ModuleImage>) -> Self {
+        // New not in old.
+        let loaded: Vec<_> = new
+            .iter()
+            .filter(|(nva, n)| {
+                old.iter()
+                    .find(|(iva, i)| nva == iva && n.path() == i.path())
+                    .is_none()
+            })
+            .map(|(va, i)| (*va, i.clone()))
+            .collect();
+
+        // Old not in new.
+        let unloaded: Vec<_> = old
+            .iter()
+            .filter(|(iva, i)| {
+                new.iter()
+                    .find(|(nva, n)| nva == iva && n.path() == i.path())
+                    .is_none()
+            })
+            .map(|(va, i)| (*va, i.clone()))
+            .collect();
+
+        Self { loaded, unloaded }
+    }
+}
+
+#[derive(Clone, Debug, Default)]
+pub struct Breakpoints {
+    saved: BTreeMap<u64, u8>,
+}
+
+impl Breakpoints {
+    pub fn set(&mut self, tracee: &mut Tracee, va: u64) -> Result<()> {
+        // Return if the breakpoint exists. We don't want to conclude that the
+        // saved instruction byte was `0xcc`.
+        if self.saved.contains_key(&va) {
+            return Ok(());
+        }
+
+        let mut data = [0u8];
+        tracee.read_memory_mut(va, &mut data)?;
+        self.saved.insert(va, data[0]);
+        tracee.write_memory(va, &[0xcc])?;
+
+        Ok(())
+    }
+
+    pub fn clear(&mut self, tracee: &mut Tracee, va: u64) -> Result<bool> {
+        let data = self.saved.remove(&va);
+
+        let cleared = if let Some(data) = data {
+            tracee.write_memory(va, &[data])?;
+            true
+        } else {
+            false
+        };
+
+        Ok(cleared)
+    }
+}
+
+#[derive(Clone, Copy, Debug, Deserialize, Eq, PartialEq, Serialize)]
+pub struct Block {
+    pub offset: u64,
+    pub size: u64,
+}
+
+type ElfFile<'a> = elf::ElfFile64<'a, LE>;
+type ElfSymbol<'a> = elf::ElfSymbol64<'a, 'a, LE>;
+type ElfSection<'a> = elf::ElfSection64<'a, 'a, LE>;
+
+pub fn find_module_blocks(module: &Path) -> Result<Vec<Block>> {
+    use object::*;
+
+    let data = std::fs::read(module)?;
+    let elf = ElfFile::parse(&data)?;
+
+    let load_va = elf.segments().map(|s| s.address()).min().unwrap();
+
+    let mut blocks = vec![];
+
+    for sym in elf.symbols() {
+        if sym.kind() == object::SymbolKind::Text && sym.size() > 0 {
+            if let Some(idx) = sym.section().index() {
+                let section = elf.section_by_index(idx)?;
+                let sym_blocks = find_symbol_blocks(section, sym)?;
+                blocks.extend(sym_blocks);
+            }
+        }
+    }
+
+    for block in &mut blocks {
+        block.offset -= load_va;
+    }
+
+    Ok(blocks)
+}
+
+pub fn find_symbol_blocks(section: ElfSection, sym: ElfSymbol) -> Result<Vec<Block>> {
+    use iced_x86::*;
```

Comment:
Why are you importing here, instead of at a the file level?

Reply:
Dev artifact. I'll do a style pass on any local/wildcard imports.

Reply:
Done, kept a few local enum imports.